dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/SelectGuestbookPage.java

```diff
@@ -0,0 +1,165 @@
+import javax.faces.application.FacesMessage;
+import javax.faces.view.ViewScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.validation.ConstraintViolation;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static edu.harvard.iq.dataverse.util.JsfHelper.JH;
+
+@ViewScoped
+@Named("selectGuestbookPage")
+public class SelectGuestbookPage implements java.io.Serializable {
+
+    private static final Logger logger = Logger.getLogger(EditDatasetMetadataPage.class.getCanonicalName());
+
+    @EJB
+    private DatasetServiceBean datasetService;
+
+    @Inject
+    private PermissionsWrapper permissionsWrapper;
+
+    @Inject
+    private DataverseRequestServiceBean dvRequestService;
+
+    @EJB
+    EjbDataverseEngine commandEngine;
+
+    private String persistentId;
+    private Long datasetId;
+
+    private Dataset dataset;
+    private DatasetVersion workingVersion;
+    private DatasetVersion clone;
+    private Guestbook selectedGuestbook;
+
+    // -------------------- GETTERS --------------------
+
+
+    public String getPersistentId() {
+        return persistentId;
+    }
+
+    public Long getDatasetId() {
+        return datasetId;
+    }
+
+    public Dataset getDataset() {
+        return dataset;
+    }
+
+    public DatasetVersion getWorkingVersion() {
+        return workingVersion;
+    }
+
+    public Guestbook getSelectedGuestbook() {
+        return selectedGuestbook;
+    }
+
+    // -------------------- LOGIC --------------------
+    public String init() {
+        if (persistentId != null) {
+            dataset = datasetService.findByGlobalId(persistentId);
+        } else if (datasetId != null) {
+            dataset = datasetService.find(datasetId);
+        }
+
+        if (dataset == null) {
+            return permissionsWrapper.notFound();
+        }
+
+        // Check permisisons
+        if (!permissionsWrapper.canUpdateDataset(dvRequestService.getDataverseRequest(), dataset)) {
+            return permissionsWrapper.notAuthorized();
+        }
+        if (!dataset.getLocks().isEmpty()) {
+            return permissionsWrapper.notAuthorized();
+        }
+
+
+        workingVersion = dataset.getEditVersion();
+        clone = workingVersion.cloneDatasetVersion();
+
+        JH.addMessage(FacesMessage.SEVERITY_INFO,
+                BundleUtil.getStringFromBundle("dataset.message.editTerms.label"),
+                BundleUtil.getStringFromBundle("dataset.message.editTerms.message"));
+
+        return StringUtils.EMPTY;
+    }
+
+    public String save() {
+
+        // Validate
+        Set<ConstraintViolation> constraintViolations = workingVersion.validate();
+        if (!constraintViolations.isEmpty()) {
+             JH.addMessage(FacesMessage.SEVERITY_ERROR, BundleUtil.getStringFromBundle("dataset.message.validationError"));
+
+            return StringUtils.EMPTY;
+        }
```

Comment:
Actually you don't have anything to validate here.

Reply:
What do you mean?

Reply:
On this page the only thing that we do is setting guestbook reference in dataset.
We don't change any data. And the only way this validation could fail is the case when dataset would be not valid even before we get it from db